core/src/main/java/org/elasticsearch/index/query/FuzzyQueryBuilder.java

```diff
@@ -151,6 +155,7 @@ public FuzzyQueryBuilder(String fieldName, boolean value) {
      * @param value The value of the term
      */
     public FuzzyQueryBuilder(String fieldName, Object value) {
+        DEPRECATION_LOGGER.deprecated("{} query is deprecated. Instead use the [match] query with fuzziness parameter", NAME);
```

Comment:
I'm quite happy to use docs tests to verify that the code is correct in lots of cases, mostly the cases where you are going to document the feature anyway and there isn't a lot of value having all the clients test it. Then you get docs and you get tests. I don't think you should add useless docs just because they add tests, but I think we could do better to document more things are if that lets us skip adding yaml tests for them then great, incentive to do the right thing.

I like the yaml tests (and junit tests with rest client) to do more complex edge case stuff that you aren't going to document.